hollow/src/main/java/com/netflix/hollow/api/producer/AbstractHollowProducer.java

```diff
@@ -419,6 +424,8 @@ long runCycle(
 
                 log.info("Populate stage completed with no delta in output state; skipping publish, announce, etc.");
             }
+            // increment the cycle count only for success case i.e with or without any detected changes
+            cycleCountWithPrimaryStatus ++;
```

Comment:
Wondering if this should happen in the `finally` block i.e. we should increment it even for failed cycles. Is there a reason to not do that?

Reply:
I think we discussed this offline about incrementing it only for successful cycles so that we don't have to account for different failure modes - esp publish blob or announce failure where a producer may lose primary status. We have guardrails in those methods for checking primary status but logically it would make more sense to just count successful cycles. Less probable but by having this here, we don't have to worry about wrong restore leading to validation failures and exception cases in populate stage? If cycle failed, we do a consistency check again on restore. Also, this would be applicable to future reporting use cases.

Reply:
But the benefit of counting both success and failed cycles i.e. moving the increment to the finallly block is that for the current feature i.e. account for eventual consistency on the first cycle since primary, we dont care if the cycle was success or fail- and if we only count success cycles then we potentially account for eventual consistency on multiple cycles before the first success cycle - seems unnecessary. 

Also thinking that we can drop these as they are redundant
```
// Logically, no new producer will resume from a mid-cycle point of a previous cycle but resetting
// this value here so that the caller code reads the correct value at all times
cycleCountSincePrimaryStatus = 0;
```
as the cycle count would get reset at the top of the next cycle anyway. The thing about 
>the caller code reads the correct value at all times
I dont know if thats entirely true, because in the success case the producer can give up primary status willingly and there will be a duration before the next invocation where the count has not been reset.

Reply:
Okay. Yes, waiting on restore due to cycle stage failures would be redundant in few scenarios. 

Re - resets in announce and announce where we check primary status are "extra" and not strictly needed given the way runCycle is invoked and works. I was trying to reset the count wherever the primary status is lost so that anytime after this a read call would be able to see the latest value but as you mentioned, this doesn't cover the case of graceful release of primary status.

Yes, you're right. RN we don't have a way to determine when a graceful give up of primary status happens (would need some tweaking on NfCuratorSingleProducer level. Will drop these.

Reply:
So could i keep the condition but change the comment to account for the 'graceful give up of primary status' case?

Reply:
Discussed offline. We wouldn't need to care about resetting the value in publish/announce. Worst case, a producer loses primary status, waits for consistency check and runs cycle to eventually fail. Another instance (if primary) will start from clean slate.